flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/sink/KafkaRecordSerializationSchemaBuilder.java

```diff
@@ -0,0 +1,270 @@
+     */
+    public KafkaRecordSerializationSchemaBuilder<IN> withValueSerializer(
+            Class<? extends Serializer<IN>> valueSerializer) {
+        checkState(valueSerializationSchema == null, exceedsValueSerializerConfigurations());
+        this.valueSerializationSchema =
+                new KafkaSerializerWrapper<>(valueSerializer, topic, topicSelector);
+        return this;
+    }
+
+    /**
+     * Sets a configurable Kafka {@link Serializer} and pass a configuration to serialize incoming
+     * elements to the value of the {@link ProducerRecord}.
+     *
+     * @param valueSerializerWithConfiguration
+     * @param configuration
+     * @param <T> type of the used serializer class
+     * @return {@link KafkaRecordSerializationSchemaBuilder}
+     */
+    public <T extends Configurable & Serializer<IN>>
+            KafkaRecordSerializationSchemaBuilder<IN> withConfigurableValueSerializer(
+                    Class<T> valueSerializerWithConfiguration, Map<String, String> configuration) {
+        checkState(valueSerializationSchema == null, exceedsValueSerializerConfigurations());
+        this.valueSerializationSchema =
+                new KafkaSerializerWrapper<>(
+                        valueSerializerWithConfiguration, configuration, topic, topicSelector);
+        return this;
+    }
+
+    /**
+     * Constructs the {@link KafkaRecordSerializationSchemaBuilder} with the configured properties.
+     *
+     * @return {@link KafkaRecordSerializationSchema}
+     */
+    public KafkaRecordSerializationSchema<IN> build() {
+        checkState(valueSerializationSchema != null, "No value serializer is configured.");
+        return new KafkaRecordSerializationSchema<>() {
+
+            @Override
+            public void open(
+                    SerializationSchema.InitializationContext context, KafkaSinkContext sinkContext)
+                    throws Exception {
+                valueSerializationSchema.open(context);
+                if (keySerializationSchema != null) {
+                    keySerializationSchema.open(context);
+                }
+                if (partitioner != null) {
+                    partitioner.open(
+                            sinkContext.getParallelInstanceId(),
+                            sinkContext.getNumberOfParallelInstances());
+                }
+            }
+
+            @Override
+            public ProducerRecord<byte[], byte[]> serialize(
+                    IN element, KafkaSinkContext context, Long timestamp) {
+                final String targetTopic = topic == null ? topicSelector.apply(element) : topic;
+                final byte[] value = valueSerializationSchema.serialize(element);
+                byte[] key = null;
+                if (keySerializationSchema != null) {
+                    key = keySerializationSchema.serialize(element);
+                }
+                final OptionalInt partition =
```

Comment:
`ProducerRecord` accepts `null` on all optional parameters, so I think we can simplify all cases to one.